lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -232,6 +239,8 @@ def edit_user(self, user, name, group, password, email):
 
         # try to update user
         try:
+            if self.module.params['update_password'] == 'on_create':
```

Comment:
if you do it before the compare, you will not need to change the compare function, password will be none

Reply:
I think the easiest place to add it will be in main before we call `manageiq_user.edit_user` , and make sure password is None before be use the funtion if we are "on_create".
That way, it will be the only change needed ( no changes to any other function needed)
And it will also be clear to someone reading the code, that we ignore the password param on edit when the "update_password" in "on_create"